src/plugin/ipc/file/fileconnection.cpp

```diff
@@ -702,15 +708,17 @@ void FileConnection::refill(bool isRestart)
   }
 
   errno = 0;
-  if (jalib::Filesystem::FileExists(_path) &&
-      stat(_path.c_str() ,&statbuf) == 0 && S_ISREG(statbuf.st_mode)) {
-    if (_offset <= statbuf.st_size && _offset <= _st_size) {
-      JASSERT(lseek(_fds[0], _offset, SEEK_SET) == _offset)
-        (_path) (_offset) (JASSERT_ERRNO);
-      //JTRACE("lseek(_fds[0], _offset, SEEK_SET)") (_fds[0]) (_offset);
-    } else if (_offset > statbuf.st_size || _offset > _st_size) {
-      JWARNING(false) (_path) (_offset) (_st_size) (statbuf.st_size)
-        .Text("No lseek done:  offset is larger than min of old and new size.");
+  if(!_ckpted_file || (_ckpted_file && !overwrite_file)) {
```

Comment:
Two issues:
 - No need for such a complex expression; the expression you have is equivalent to: `!(ckpted_file && overwrite_file)`
 - Unless I'm reading this incorrectly, this new expression disallows seeking to the original offset when a file was ckpt-ed and overwritten by its ckpt-ed copy. I don't think you want that.